File: wc/go-wc/pkg/counter/counter.go
package counter

import (
	"fmt"
	"strings"
	"unicode/utf8"
)

func Count(countBytes, countLines, countWords, countCharacters bool, content string) string {
	values := make([]string, 0)
	if countBytes {
		nBytes := fmt.Sprint(countContentBytes(content))
		values = append(values, nBytes)
	}

	if countLines {
		lines := fmt.Sprint(countContentLines(content))
		values = append(values, lines)
	}

	if countWords {
		words := fmt.Sprint(countContentWords(content))
		values = append(values, words)
	}

	if countCharacters {
		characters := fmt.Sprint(countContentCharacters(content))
		values = append(values, characters)
	}

	if !countBytes && !countLines && !countWords && !countCharacters {
		nBytes := fmt.Sprint(countContentBytes(content))
		lines := fmt.Sprint(countContentLines(content))
		words := fmt.Sprint(countContentWords(content))
		characters := fmt.Sprint(countContentCharacters(content))
		values = append(values, nBytes, lines, words, characters)
	}

	return strings.Join(values, "\t")
}

func countContentBytes(content string) int {
	return len([]byte(content))
}

func countContentLines(content string) int {
	return len(strings.Split(content, "\r\n"))
}

func countContentWords(content string) int {
	return len(strings.Fields(content))
}

func countContentCharacters(content string) int {
	return utf8.RuneCountInString(content)
}

File: wc/node-wc/counter.js

const countBytes = (input) => {
    return Buffer.byteLength(input, 'utf-8')
}

const countLines = (input) => {
    return input.split('\n').length;
}

const countWords = (input) => {
    return input.trim().split(/\s+/).length;
}

const countCharacters = (input) => {
    return input.length
}

module.exports = {
    processInput: (input, options) => {
        let response = "";
        
        if (options.c) {
            const bytes = countBytes(input)
            response += `\t${bytes}`
        }

        if (options.l) {
            const lines = countLines(input)
            response += `\t${lines}`  
        }

        if (options.w) {
            const words = countWords(input)
            response += `\t${words}`
        }

        if (options.m) {
            const characters = countCharacters(input)
            response += `\t${characters}`
        }

        if (!options.c && !options.l && !options.w && !options.m) {
            const bytes = countBytes(input)
            const lines = countLines(input)
            const words = countWords(input)
            const characters = countCharacters(input)
            response = `\t${bytes}\t${lines}\t${words}\t${characters}`
        }

        if (!options.isFromStdin) {
            response += `\t${options.filePath}`
        }

        console.log(response)
    }
}